.github/scripts/promote_branch.sh

```diff
@@ -90,6 +101,144 @@ check_if_any_commits_in_last_week() {
     fi
 }
 
+# GraphQL function to fetch PRs since a given date
+fetch_prs_graphql() {
+    local since_date="$1"
+    local page_size="${2:-${GITHUB_GRAPHQL_PAGE_SIZE:-100}}"  # Configurable via env var
+    local cursor="${3:-null}"
+
+    # Calculate cursor parameter for pagination
+    local cursor_param=""
+    if [ "$cursor" != "null" ]; then
+        cursor_param=", after: \\\"$cursor\\\""
+    fi
+
+    # GraphQL query to fetch PRs with commits
+    local query_string="query { repository(owner: \\\"$ORG\\\", name: \\\"$REPO\\\") { pullRequests(first: $page_size, states: MERGED, orderBy: {field: UPDATED_AT, direction: DESC}$cursor_param) { pageInfo { hasNextPage endCursor } nodes { number title url mergedAt mergeCommit { oid } labels(first: 20) { nodes { name color description } } commits(first: 250) { nodes { commit { oid messageHeadline } } } } } } }"
+    local query="{\"query\": \"$query_string\"}"
+
+    # Make GraphQL request
+    local response=$(curl -s -X POST \
```

Comment:
It would probably be good to do some basic error handling - graphql will return errors in the json response. Maybe fail if .errors is non-null?